src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -494,92 +481,53 @@ def _special_cases(self, text, cmd, outside):
-                        cmd_base = cmd_base.replace(SELECT_SYMBOL['query'], result, 1)
-                    self.cli_execute(cmd_base)
-                    continue_flag = True
-                elif all(isinstance(result, list) for result in results):
-                    if len(results) > 1:
-                        for res_counter, _ in enumerate(results[0]):
-                            base = cmd_base
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    else:
-                        for res_counter, _ in enumerate(results[0]):
-                            base = cmd_base
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    continue_flag = True
-
+            if len(args) == 1:  # if arguments start with query_symbol, just print query result
+                if args[0] == query_symbol:
+                    print(json.dumps(self.last.result, sort_keys=True, indent=2), file=self.output)
+                elif len(args) == 1 and args[0].startswith(query_symbol):
+                    query = args[0][symbol_len:]
+                    print(json.dumps(jmespath.search(query, input_dict), sort_keys=True, indent=2), file=self.output)
+            else:  # query, inject into cmd
+                def sub_result(arg):
+                    escaped_symbol = re.escape(query_symbol)
+
+                    def jmespath_query(match):
+                        query_result = jmespath.search(match.group(0)[symbol_len:], input_dict)
```

Comment:
`match.group(0)` can be null if there's no match for the regex right?
Then should handle that case.

Reply:
this wont happen since re.sub will only call jmespath_query() on each match it finds. If there's no match, it just wont do anything.